models/hl7-v3/dispense-notification.ts

```diff
@@ -150,15 +154,46 @@ export class DispenseNotificationSuppliedLineItem {
   component: Array<DispenseNotificationSuppliedLineItemComponent>
   component1: DispenseNotificationSuppliedLineItemComponent1
   pertinentInformation3: dispenseCommon.SuppliedLineItemPertinentInformation3
+  pertinentInformation2: SupplyPertinentInformation2
   inFulfillmentOf: dispenseCommon.SuppliedLineItemInFulfillmentOf
 
-  constructor(id: codes.GlobalIdentifier) {
+  constructor(id: codes.GlobalIdentifier, pertInfo2 : SupplyPertinentInformation2) {
     this.id = id
     this.code = new codes.SnomedCode("225426007", "Administration of therapeutic substance (procedure)")
-    this.effectiveTime = core.Null.NOT_APPLICABLE
+    this.effectiveTime = core.Null.NOT_APPLICABLE,
+    this.pertinentInformation2 = pertInfo2
   }
 }
 
+
+ export type SupplyPertinentInformation2 = PertinentInformation2 | PertinentInformation2NonDispensing
+
+
+export class PertinentInformation2 implements ElementCompact {
+  _attributes: core.AttributeTypeCode & core.AttributeContextConductionInd = {
+    typeCode: "PERT",
+    contextConductionInd: "true"
+  }
+
+  seperatableInd: core.BooleanValue
+
+  constructor() {
+    this.seperatableInd = new core.BooleanValue(true)
+  }
+}
+
+export class PertinentInformation2NonDispensing extends PertinentInformation2 {
+
+  readonly pertientNonDispensingReason: NonDispensingReason
+
+  constructor(pertientNonDispensingReason : NonDispensingReason) {
+    super();
```

Comment:
Could we use the same pattern as the other classes? From a quick look we tend to do something like:
```
export class PertinentInformation2 implements ElementCompact {
  ...
  constructor(nonDispensingReason?: PertinentInformation2NonDispensing) { ... }
}

const nonDispensingReason = new PertinentInformation2NonDispensing(...)
const perinentInformation2 = new PertinentInformation2(nonDispensingReason)
```
There should be something similar here:
https://github.com/NHSDigital/electronic-prescription-service-api/pull/997/files#diff-262a137ba9c4a2aadb906ded8e0e38e42ae8aace598236f197a7ccdfcb21a84aR128-R132

Reply:
In the case where the nonDispensingReason is null then output XML would have a single tag <pertient nonDispensingReason/>.